langroid/language_models/azure_openai.py

```diff
@@ -98,6 +90,42 @@ def __init__(self, config: AzureConfig):
                 please set it to chat model name in your deployment."""
             )
 
+        if (
+            self.config.azure_openai_client_provider
+            and self.config.azure_openai_async_client_provider
```

Comment:
Might be good to separate these two, i.e. if only one of them is specified, use that one for the client. If you think we should enforce that either none or both are specified, it may be good to raise an error with a friendly msg saying that only one of these was specified, and the other must also be specified.

Reply:
Yeah, that's a good point. I've changed it now so you only need to supply one, but if you only supply one, a warning will be given.